src/modules/accommodations/accommodations.service.ts

```diff
@@ -63,6 +73,99 @@ export class AccommodationsService {
     };
   }
 
+  async createAccommodation(
+    createDto: CreateAccommodationDto,
+    userId: string,
+  ): Promise<Accommodation> {
+    const { amenities, categoryId, ...accommodationData } = createDto;
+    const allAmenities = await this.amenitiesService.getAllAmenities();
+    const resolvedAmenities = amenities
+      ? allAmenities.filter((amenity) => amenities.includes(amenity.id))
+      : [];
```

Comment:
Instead of this, you can just get only those amenities that are included in the array from directly database, using `where in` query.